### PR TITLE
[New Nav] Implement dark theme for the Colony Sidebar

### LINKED ISSUE #3049: [New Navigation] Update dark theme for Sidebar
## Description

The new Sidebar component wasn't built with dark mode in mind so its colours are all over the place when dark mode is enabled. We need to make it match the dark mode Figma designs:

## Acceptance Criteria

* [ ] Should match the [Desktop Figma designs](https://www.figma.com/design/uKxVl441Wnn02SNiOA24De/Navigation?node-id=4663-19803&t=kg19tGXpBCht52mI-4)

Image

* [ ] Should match the [Mobile Figma designs](https://www.figma.com/design/uKxVl441Wnn02SNiOA24De/Navigation?node-id=4663-19996&t=kg19tGXpBCht52mI-4)

Image

### PR DESCRIPTION
**Pull request by rumzledz**
## Description

Screenshot 2024-09-17 at 17 19 04

## Testing

1. Toggle on dark mode
2. Verify that the Colony Sidebar matches the [Desktop dark mode designs](https://www.figma.com/design/uKxVl441Wnn02SNiOA24De/Navigation?node-id=4663-19800&t=cNVkt1qzBSbIs3yI-4)
3. Verify that the Colony Sidebar matches the [Mobile dark mode designs](https://www.figma.com/design/uKxVl441Wnn02SNiOA24De/Navigation?node-id=4663-19996&t=cNVkt1qzBSbIs3yI-4)

Resolves #3049

### CODE REVIEW

File: src/components/common/Onboarding/Wizard/WizardSidebarItem.tsx [hunk 1]
```diff
@@ -41,12 +45,22 @@ const WizardSidebarItem = ({
         <div
           className={clsx('h-2.5 w-2.5 rounded-full', {
             'bg-base-white':
-              (currentStep >= stepId && hasSubItems) ||
-              (!hasSubItems && currentStep === stepId),
-            'border border-base-white': currentStep < stepId,
+              ((currentStep >= stepId && hasSubItems) ||
+                (!hasSubItems && currentStep === stepId)) &&
```
Comment (iamsamgibbs) [inline src/components/common/Onboarding/Wizard/WizardSidebarItem.tsx:49]: Maybe we could make a `const isCompletedStep` or something

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Looking good! The only issue I could find is buttons not staying highlighted whilst you are on the page.

Screenshot 2024-09-24 at 09 44 37

Otherwise looks good!

Screenshot 2024-09-24 at 09 48 59

Screenshot 2024-09-24 at 09 49 24

**Comment by rumzledz** [comment]
> Looking good! The only issue I could find is buttons not staying highlighted whilst you are on the page

@iamsamgibbs Whoops! 👀

File: src/components/common/Onboarding/Wizard/WizardSidebarItem.tsx (hunk 1 above)
Comment (rumzledz) [reply src/components/common/Onboarding/Wizard/WizardSidebarItem.tsx:49]: All right will do :)

**Comment by rumzledz** [comment]
@iamsamgibbs pushed! 🚀

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
The active menu item is looking good now:

Screenshot 2024-09-24 at 11 20 18

Unfortunately it has introduced another bug on the sub-menu item:

Screenshot 2024-09-24 at 11 20 26

**Review by mmioana — CHANGES_REQUESTED** [review]
Hey @rumzledz thanks for all your hard work on this issue 🙌 💯 

From testing your PR, I noticed there is an issue in the dark mode for the `Finances` navigation item, though only on desktop. I'll leave this recording to better understand the problem

https://github.com/user-attachments/assets/a3031fb1-c67c-414a-bf09-988b3a8bbcc6

I assume there are some changes needed in `src/components/v5/shared/Navigation/Sidebar/partials/SidebarRouteItem/SidebarRouteItem.tsx`

However on mobile it works as expected

https://github.com/user-attachments/assets/d15a5c16-3eef-4cc2-80f2-859e46b5504a

Also the colony switcher looks good on dark mode, though I couldn't find the design to compare 😓 
Screenshot 2024-09-24 at 12 22 10

Thanks again for all your effort 🚀

**Comment by rumzledz** [comment]
Ah damn.. all right I'll fix it up @iamsamgibbs @mmioana thank you!

**Comment by rumzledz** [comment]
All right pushed again! 😂 @iamsamgibbs @mmioana @bassgeta

**Review by bassgeta — APPROVED** [review]
Nicely done tackling all the edge cases in this PR, it's looking more and more complete with each PR!
Uncollapsed light mode:
image
Collapsed:
image
Dark mode expanded:
image
Collapsed:
image

Without spamming to many pictures, the mobile also behaves correctly and the correct route always gets highlighted

**Review by mmioana — APPROVED** [review]
You're on fire, smashing all this nav issues @rumzledz 🔥 

Light mode is unaffected
Screenshot 2024-09-25 at 09 20 26

Dark mode
Screenshot 2024-09-25 at 09 20 49
Screenshot 2024-09-25 at 09 20 55
Screenshot 2024-09-25 at 09 20 58
Screenshot 2024-09-25 at 09 21 53

Dark mode on mobile
Screenshot 2024-09-25 at 09 21 21
Screenshot 2024-09-25 at 09 21 26
Screenshot 2024-09-25 at 09 21 33

Creating a new colony
Screenshot 2024-09-25 at 09 22 50
Screenshot 2024-09-25 at 09 23 01
Screenshot 2024-09-25 at 09 23 11

No wallet connected
Screenshot 2024-09-25 at 09 23 27

No colonies joined
Screenshot 2024-09-25 at 09 23 52

Let's merge!!! 🎉

**Review by iamsamgibbs — APPROVED** [review]
Looking great now! Time to merge! 🚀 

Screenshot 2024-09-25 at 08 40 50
Screenshot 2024-09-25 at 08 41 13
Screenshot 2024-09-25 at 08 41 33

**Pull request merged**